lib/identifiers/doi.rb

```diff
@@ -33,10 +33,10 @@ def self.extract(str)
         .to_s
         .downcase
         .scan(PATTERN)
-        .map { |doi|
-          next doi if doi =~ VALID_ENDING
+        .flat_map { |doi|
+          next [doi] if doi =~ VALID_ENDING
```

Comment:
I prefer the `extract_one` implementation in the comment, the other one was slightly confusing regarding the `flat_map` and the need to return an array instead for the recursion